src/main/java/org/nuxeo/ecm/platform/pdf/service/PDFTransformationServiceImpl.java

```diff
@@ -0,0 +1,247 @@
+        }
+    }
+
+    @Override
+    public Blob overlayPDF(Blob input, Blob overlayBlob) {
+        try (PDDocument pdfDoc = PDDocument.load(input.getStream());
+             PDDocument pdfOverlayDoc = PDDocument.load(overlayBlob.getStream())) {
+            Overlay overlay = new Overlay();
+            overlay.overlay(pdfOverlayDoc, pdfDoc);
+            return saveInTempFile(pdfDoc);
+        } catch (IOException | COSVisitorException e) {
+            throw new NuxeoException(e);
+        }
+    }
+
+    public  Point2D computeTranslationVector(double pageWidth, double watermarkWidth,
+                                               double pageHeight, double watermarkHeight,
+                                               WatermarkProperties properties) {
+        double xTranslation;
+        double yTranslation;
+        double xRotationOffset = 0;
+        double yRotationOffset = 0;
+
+        if (properties.getRotation() != 0) {
+            Rectangle2D rectangle2D =
+                    new Rectangle2D.Double(
+                            0, -watermarkHeight, watermarkWidth, watermarkHeight);
+            AffineTransform at = AffineTransform.getRotateInstance(
+                    -Math.toRadians(properties.getRotation()), 0, 0);
+            Shape shape = at.createTransformedShape(rectangle2D);
+            Rectangle2D rotated = shape.getBounds2D();
+
+            watermarkWidth = rotated.getWidth();
+            if (!properties.isInvertX() || properties.isRelativeCoordinates()) {
+                xRotationOffset = -rotated.getX();
+            } else {
+                xRotationOffset = rotated.getX();
+            }
+
+            watermarkHeight = rotated.getHeight();
+            if (!properties.isInvertY() || properties.isRelativeCoordinates()) {
+                yRotationOffset = rotated.getY()+rotated.getHeight();
+            } else {
+                yRotationOffset = -(rotated.getY()+rotated.getHeight());
+            }
+        }
+
+        if (properties.isRelativeCoordinates()) {
+            xTranslation = (pageWidth - watermarkWidth ) * properties.getxPosition() + xRotationOffset;
+            yTranslation = (pageHeight - watermarkHeight ) * properties.getyPosition() + yRotationOffset;
+        } else {
+            xTranslation = properties.getxPosition() + xRotationOffset;
+            yTranslation = properties.getyPosition() + yRotationOffset;
+            if (properties.isInvertX()) xTranslation = pageWidth - watermarkWidth - xTranslation;
+            if (properties.isInvertY()) yTranslation = pageHeight - watermarkHeight - yTranslation;
+        }
+        return new Point2D.Double(xTranslation, yTranslation);
+    }
+
+    protected FileBlob saveInTempFile(PDDocument PdfDoc) throws IOException, COSVisitorException {
+        File tempFile = Framework.createTempFile("nuxeo-pdfutils-", ".pdf");
+        PdfDoc.save(tempFile);
+        return new FileBlob(tempFile,MIME_TYPE,tempFile.getName());
```

Comment:
This temporary file is not properly tracked for deletion when not referenced anymore. You should do:

    Blob blob = Blobs.createBlobWithExtension(".pdf"); // creates a tracked temporary file
    blob.setMimeType(MIME_TYPE);
    PdfDoc.save(blob.getFile());
    return blob;